library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -124,18 +105,16 @@ impl<T: Clone, A: Allocator + Clone> Clone for VecDeque<T, A> {
     }
 
     fn clone_from(&mut self, other: &Self) {
```

Comment:
This will need to reallocate if `other.len() > self.capacity()`, so consider adding a check here like
```rust
if other.len() > self.capacity() {
    return *self = self.clone();
}
```
because there's no point in trying to re-use the buffer in that situation, thus we might as well simplify it.

Reply:
If a larger allocation is needed it might be better to drop the old allocation first (by shrinking it to 0) so it gets returned to the allocator before requesting a new one, that way it has a chance to reuse it.

Some `Allocator::grow` impls already try to reuse it, but they can't gurantee it because there may not be enough room and address space remapping shenanigans only work for large allocations, not for pooled small ones.

Reply:
In that case, wouldn't it be smarter to do that in `RawVec` itself? That way, `Vec` could also take advantage of it.

Reply:
`grow` is a good point; this might as well stay as-is.